utilities.py

```diff
@@ -171,10 +174,319 @@ def closest(start: list, test_pts: list):
 
     return closest_pt, distance_minimum
 
+
 def simulate_sensors(environment, sensors):
     '''Simulate a list of sensors'''
 
     ROBOT = environment.get("ROBOT", None)
     for d_id in sensors:
         if d_id in ROBOT.sensors:
             ROBOT.sensors[d_id].simulate(0, environment)
+
+
+def angle(segment1:list, segment2:list):    
+
+    theta1 = math.degrees(math.atan2(segment1[0][1]-segment1[1][1],segment1[0][0]-segment1[1][0]))
+    theta2 = math.degrees(math.atan2(segment2[0][1]-segment2[1][1],segment2[0][0]-segment2[1][0]))
+
+    diff = abs(theta1 - theta2)
+    
+    # while diff>=360:
+    #     diff-=360
+    
+    if diff>270:
+        return 360-diff
+    if diff>180:
+        return diff - 180
+    if diff>90:
+        return 180 - diff
+
+
+def closestFast(start: list, test_pts: list):
```

Comment:
for Python variables and function names, convention is to use `snake_case` rather than `camelCase`.